### ISSUE TITLE
do it work with slim 4? 

### ISSUE THREAD

**Issue by josebrc** [chadicus/slim-oauth2#16]

**Comment by iCaspar** [comment]
The default branch doesn't work with Slim 4, owing (at least) to change in Middleware Interfaces required by Slim in v4. I see there has been some work on a v4 branch in the [middleware repo](https://github.com/chadicus/slim-oauth2-middleware/tree/4.x), but nothing since last year.
@chadicus , is this still something you're maintaining (and if so, is there anything that will help/contribute), or should we better fork this and proceed on our own? Much appreciated!

**Comment by chadicus** [comment]
@iCaspar @joseL02 yes, i am still maintaining it.  However it is on a part time basis. I was trying to get v4 ready in time for the Slim v4 release, but i was not able to do so.
Pull requests are welcome.

**Comment by chadicus** [comment]
Obviously i was not able to get the upgrade done as promised. Any help would be greatly appreciated.
